src/chat/messageTypes.ts

```diff
@@ -0,0 +1,33 @@
+import { LanguageModelChatMessage } from "vscode";
+import { PARTICIPANT_ID } from "./constants";
+
+/**
+ * Returns a {@link LanguageModelChatMessage.User User message} with a "user" tag to clearly
+ * distinguish it from other `User` message implementations.
+ *
+ * This should only be used for the most recent {@link ChatRequest.prompt} and any historic messages.
+ */
+export function userMessage(message: string) {
+  return LanguageModelChatMessage.User(message, "user");
+}
+
+/**
+ * Returns an {@link LanguageModelChatMessage.Assistant Assistant message} with a participant tag as
+ * the {@linkcode PARTICIPANT_ID} to clearly separate it from other `Assistant` message implementations.
+ */
+export function participantMessage(message: string) {
+  return LanguageModelChatMessage.Assistant(message, PARTICIPANT_ID);
+}
+
+// TODO: update this if the `LanguageModelChatMessage` API changes and a `System` message is added
+/**
+ * Prefix and tag a {@link LanguageModelChatMessage.User User message} message to treat as a
```

Comment:
maybe we should mention [why this is needed](https://github.com/microsoft/vscode-extension-samples/blob/main/chat-sample/src/toolsPrompt.tsx#L83)

Reply:
Added [here](https://github.com/confluentinc/vscode/pull/1478/commits/8f347f3881ea23c65e2b72aa84489ac845c96481) since it's at the bottom of the tool handling